iroha_p2p/src/peer.rs

```diff
@@ -375,42 +411,88 @@ where
             "Starting actor for connection with peer {}",
             &self.id.address
         );
-        let mut count = 0;
+        self.broker.subscribe::<StopSelf, _>(ctx);
+    }
+}
+
+#[async_trait::async_trait]
+impl<T, K, E> ContextHandler<Connect> for Peer<T, K, E>
```

Comment:
I thought that we just connected on start of peer...

Reply:
No, `on_start()` must finish quickly for `Network` to get connection id from the peer and save it in a hashmap. Then, we command the peer to start connection. If we don't save the peer before connection is done we can't find that peer in `impl<T, K, E> Handler<PeerMessage<T>> for NetworkBase<T, K, E>`.